<?xml version="1.0"?>
<launch>

  <arg name="sim"		default="false"/>
  <arg name="camera_name"	default="realsense"/>
  <arg name="manager"		default="camera_manager"/>
  <arg name="publish_rate"	default="100"/>

  <!-- Launch MoveIt! planner -->
  <include file="$(find toyota1_bringup)/launch/calibration_move_group.launch">
    <arg name="vis"		value="$(arg sim)"/>
  </include>

  <!-- Launch cameras -->
  <include file="$(find toyota1_routines)/launch/toyota1_cameras.launch">
    <arg name="sim"		 value="$(arg sim)"/>
    <arg name="phoxi_config"	 value="$(find toyota1_routines
					)/config/motioncam.yaml"/>
    <arg name="realsense_config" value="$(find toyota1_routines
					)/config/realsense.yaml"/>
    <arg name="manager"		 value="$(arg manager)"/>
  </include>

  <!-- Start ArUco -->
  <node name="aruco_detector" pkg="nodelet" type="nodelet" output="screen"
        args="load aist_aruco_ros/aruco_detector camera_manager">
    <remap if="$(eval camera_name == 'phoxi')"
	   from="/camera_info"	to="$(arg camera_name)/camera_info"/>
    <remap if="$(eval camera_name == 'phoxi')"
	   from="/pointcloud"	to="$(arg camera_name)/pointcloud"/>
    <remap unless="$(eval camera_name == 'phoxi')"
	   from="/camera_info"	to="$(arg camera_name)/color/camera_info"/>
    <remap unless="$(eval camera_name == 'phoxi')"
	   from="/image"	to="$(arg camera_name)/color/image_raw"/>
    <remap unless="$(eval camera_name == 'phoxi')"
	   from="/depth"	to="$(arg camera_name
				    )/aligned_depth_to_color/image_raw"/>
    <rosparam file="$(dirname)/../config/$(arg camera_name).yaml"/>
  </node>

  <!-- Spawn pose_head_controller -->
  <arg  name="controller_name"	value="arm_controller"/>
  <node name="arm_pose_head_tracker"
        pkg="aist_controllers" type="pose_head_tracker"
        output="screen" respawn="false">
    <param name="publish_rate"	value="$(arg publish_rate)"/>
    <param name="base_link"	value="arm_base_link"/>
    <param name="controller"	value="/$(arg controller_name)"/>
    <remap from="/state"	to="/$(arg controller_name)/state"/>
  </node>

</launch>
